src/client/hs.rs

```diff
@@ -980,6 +980,8 @@ impl ExpectTLS13Certificate {
 }
 
 impl State for ExpectTLS13Certificate {
+    fn details(&self) -> &HandshakeDetails { &self.handshake }
```

Comment:
I think that instead of exposing these parts of the handshake layer for every handshake state, we can move the transcript "upwards" out of `HandshakeDetails` into `ClientSessionImpl`? Or, if we need to do the same for server-side support, into `SessionCommon`